src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -492,38 +351,34 @@ captureDeclaredGroupData(const Opm::Schedule&                 sched,
                          const Opm::SummaryState&             sumState,
                          const std::vector<int>&              inteHead)
 {
-    const auto indexGroupMap = currentGroupMapIndexGroup(sched, simStep, inteHead);
-    const auto nameIndexMap = currentGroupMapNameIndex(sched, simStep, inteHead);
-
-    std::vector<const Opm::Group*> curGroups(ngmaxz(inteHead), nullptr);
-
-    auto it = indexGroupMap.begin();
-    while (it != indexGroupMap.end())
-    {
-        curGroups[static_cast<int>(it->first)] = it->second;
-        it++;
+    std::vector<const Opm::Group2*> curGroups(ngmaxz(inteHead), nullptr);
+    for (const auto& group_name : sched.groupNames(simStep)) {
+        const auto& group = sched.getGroup2(group_name, simStep);
+        int ind = (group.name() == "FIELD")
+            ? ngmaxz(inteHead)-1 : group.insert_index()-1;
+        curGroups[ind] = std::addressof(group);
     }
 
     groupLoop(curGroups, [&sched, simStep, &inteHead, this]
-              (const Group& group, const std::size_t groupID) -> void
+              (const Group2& group, const std::size_t groupID) -> void
                          {
                              auto ig = this->iGroup_[groupID];
 
                              IGrp::staticContrib(sched, group, this->nWGMax_, this->nGMaxz_,
-                                                 simStep, ig, inteHead);
+                                                 simStep, ig);
```

Comment:
Removing `inteHead` here means you can (and should) remove the object from the lambda's capture list too.

Reply:
ok